docs/reference/redis.md

```diff
@@ -0,0 +1,202 @@
+
+## Availability and prerequisites
+
+This connector is available as a **connector client** from the **Python connectors framework**. To use this connector, satisfy all [connector client requirements](https://www.elastic.co/guide/en/enterprise-search/master/build-connector.html).
+
+This connector is in **beta** and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.
+
+## Usage
+
+To use this connector as a **connector client**, use the **Redis** tile from the connectors list or **Customized connector** workflow.
+
+For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).
+
+## Configuration
+
+### Configure Redis connector
+
+The following configuration fields are required:
+
+#### `host`  (required)
+
+The IP of your Redis server/cloud. Example:
+
+- `127.0.0.1`
+- `redis-12345.us-east-1.ec2.cloud.redislabs.com`
+
+#### `port`  (required)
+
+where the Redis server/cloud instance is hosted. Example:
+
+- `6379`
+
+#### `username`  (optional)
+
+Username for your Redis server/cloud. Example:
+
+- `default`
+
+#### `password`  (optional)
+
+Password for your Redis server/cloud. Example:
+
+- `changeme`
+
+#### `database`  (required)
+
+List of database index for your Redis server/cloud. * will fetch data from all databases. Example:
+
+- `0,1,2`
+- `*`
+
+*NOTE*:
+
+- This field is ignored when using advanced sync rules.
+
+## Documents and syncs
+
+The connector syncs the following objects and entities:
+- KEYS and VALUES of every database index
+
+
+*NOTE*:
+- Permissions are not synced. **All documents** indexed to an Elastic deployment will be visible to **all users with access** to that Elastic Deployment.
+
+## Sync rules
+
+[Basic sync rules](https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html#sync-rules-basic) are identical for all connectors and are available by default.
+
+## Advanced Sync Rules
+
+Advanced sync rules are defined through a source-specific DSL JSON snippet.
+
+Use advanced sync rules to filter data to be fetched from Redis server/cloud. They take the following parameters:
+
+1. `database`: Database index of Redis server/cloud. The type of value should be integer.
+2. `key_pattern`: Pattern of the key for searching in Redis server/cloud
+3. `type_`: Type of key in Redis server/cloud. Supported values are HASH, LIST, SET, STREAM, STRING, ZSET
+
+*NOTE*:
+
+- Either `key_pattern` or `type_` must be provided, but it is not necessary to include both. However, at least one of them must be present to proceed.
+
+### Advanced sync rules examples
+
+#### Fetch database records with keys starting with alpha.
```

Comment:
These titles are too long. Instead use the following pattern:

```markdown
#### Example 1

**Fetch database records where keys start with `alpha`**
```

Use this pattern for all examples. :)